### PR TITLE
PMM-11192   [DOC] PMM Client upgrade documentation for different distribution methods - Packages

### PR DESCRIPTION
**Pull request by igroene**
Improve the documentation of pmm-client upgrade process

### CODE REVIEW

File: docs/how-to/upgrade.md [hunk 1]
```diff
@@ -37,7 +37,32 @@ If one is available, click the update button to update to the version indicated.
 
 ## Updating a PMM-Agent
 
-PMM-Agent can be updated from tarball:
+### Package Manager
+
+The easiest way to manage the agent installation and updates is to rely on the operating system package manager. Use the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package to handle access to Percona repositories.
+
+The pmm agent is available in the `tools` repository. To deploy a new version of the agent, simply replace the currently installed package.
```
Comment (atymchuk) [inline docs/how-to/upgrade.md:44]:
```suggestion
PMM Agent is available from the `tools` repository. To deploy a new version of the agent, you have the following options:
- install the latest version if you are using a package manager
- replace the currently installed package with the latest if you chose to install PMM Agent from the tarball
```

File: docs/how-to/upgrade.md [hunk 2]
```diff
@@ -37,7 +37,32 @@ If one is available, click the update button to update to the version indicated.
 
 ## Updating a PMM-Agent
 
-PMM-Agent can be updated from tarball:
+### Package Manager
+
+The easiest way to manage the agent installation and updates is to rely on the operating system package manager. Use the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package to handle access to Percona repositories.
```
Comment (atymchuk) [inline docs/how-to/upgrade.md:42]:
```suggestion
The easiest way to manage the agent installation and updates is to rely on the operating system package manager. Use the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package to help you install Percona software in general, and PMM Agent in particular.
```

File: docs/how-to/upgrade.md [hunk 3]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
 
 ## Updating a PMM-Agent
 
-PMM-Agent can be updated from tarball:
+### Package Manager
+
+The easiest way to manage the agent installation and updates is to rely on the operating system package manager. Use the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package to help you install Percona software in general, and PMM Agent in particular.
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:52]:
```suggestion
The most convenient and efficient way to manage the installation and updates of the PMM Agent is by leveraging your operating system's package manager. Percona provides the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package, which helps you install Percona software, including the PMM Agent.```

File: docs/how-to/upgrade.md [hunk 4]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+
+PMM Agent is available from the `tools` repository. To deploy a new version of the agent, you have the following options:
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:54]:
```suggestion
PMM Agent is available from the `tools` repository. To deploy a new version of the Agent, you have two options:
```

File: docs/how-to/upgrade.md [hunk 5]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+- install the latest version if you are using a package manager
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:55]:
```suggestion
- If you are using a package manager, you can install the latest version of the PMM Agent directly.
```

File: docs/how-to/upgrade.md [hunk 6]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+- replace the currently installed package with the latest if you chose to install PMM Agent from the tarball
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:56]:
```suggestion
- If you initially installed the PMM Agent from a tarball, you can replace the currently installed package with the latest version.
```

File: docs/how-to/upgrade.md [hunk 7]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+
+For example the following commands install the latest version of the agent on Red Hat or derivatives:
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:58]:
```suggestion
For example, to install the latest version of the PMM Agent on Red Hat or its derivatives, you can use the following commands:
```

File: docs/how-to/upgrade.md [hunk 8]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+```
+percona-release enable tools
+yum update pmm2-client
+```
+This keeps the existing agent configuration.
```
Comment (catalinaadam) [inline docs/how-to/upgrade.md:63]:
```suggestion
By running these commands, your existing agent configuration will be preserved during the update process.
```

**Comment by igroene** [comment]
this one has been pending for too long. Can we merge it?

File: docs/how-to/upgrade.md [hunk 9]
```diff
@@ -47,7 +47,34 @@ If one is available, click the update button to update to the version indicated.
...
+
+The most convenient and efficient way to manage the installation and updates of the PMM Agent is by leveraging your operating system's package manager. Percona provides the [percona-release](https://docs.percona.com/percona-software-repositories/installing.html) package, which helps you install Percona software, including the PMM Agent.```
+
+PMM Agent is available from the `tools` repository. To deploy a new version of the Agent, you have two options:
+- If you are using a package manager, you can install the latest version of the PMM Agent directly.
+- If you initially installed the PMM Agent from a tarball, you can replace the currently installed package with the latest version.
+
+For example, to install the latest version of the PMM Agent on Red Hat or its derivatives, you can use the following commands:
+```
+percona-release enable tools
```
Comment (rnovikovP) [inline docs/how-to/upgrade.md:60]:
```suggestion
percona-release enable pmm2-client
```


@EvgeniyPatlan PTAL, is this the right repo now?

File: docs/how-to/upgrade.md [hunk 10]
```diff
@@ -38,19 +37,55 @@ The panel shows:
...
+To deploy a new version of the Agent via package manager, simply replace the currently installed package with the latest version of the PMM Agent or with a specific version.
+
+#### Install the latest PMM Agent version
+
+Run the commands below to install the latest PMM Agent version via package manager and keep your existing Agent configuration during the update process.
+
+For example, to install the latest version of the PMM Agent on Red Hat or its derivatives:
+
+    ```
+    percona-release enable tools
```
Comment (BupycHuk) [inline docs/how-to/upgrade.md:67]:
```suggestion
    percona-release enable pmm2-client
```

File: docs/how-to/upgrade.md [hunk 11]
```diff
@@ -38,19 +37,55 @@ The panel shows:
...
+
+To deploy a specific version of the PMM Agent via package manager, check the available versions and then provide the full name of the package. For example:
+
+    ```
+    yum --showduplicates search pmm2-client
+    pmm2-client-2.36.0-6.el7.x86_64 : PMM-agent
+    pmm2-client-2.37.0-6.el7.x86_64 : PMM-agent
+    pmm2-client-2.37.1-6.el7.x86_64 : PMM-agent
+
+    yum update pmm2-client-2.37.1-6.el7.x86_64
```
Comment (ademidoff) [inline docs/how-to/upgrade.md:80]:
```suggestion
    yum --showduplicates search pmm2-client
    pmm2-client-2.39.0-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.40.0-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.40.1-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.41.0-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.41.1-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.41.2-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)
    pmm2-client-2.42.0-6.el9.x86_64 : Percona Monitoring and Management Client (pmm-agent)

    yum update pmm2-client-2.41.2-6.el9.x86_64
```

**Review by ademidoff — COMMENTED** [review]
We'd rather provide EL9 based examples :)

**Pull request merged**